prototype_source/nnapi_mobilenetv2.rst

```diff
@@ -0,0 +1,173 @@
+(Prototype) Convert MobileNetV2 to NNAPI
+========================================
+
+Introduction
+------------
+
+This tutorial shows how to prepare a computer vision model to use
+`Android's Neural Networks API (NNAPI) <https://developer.android.com/ndk/guides/neuralnetworks>`_.
+NNAPI provides access to powerful and efficient computational cores
+on many modern Android devices.
+
+PyTorch's NNAPI is currently in the "prototype" phase and only supports
+a limited range of operators, but we expect to solidify the integration
+and expand our operator support over time.
+
+
+Model Preparation
+-----------------
+
+First, we must prepare our model to execute with NNAPI.
+This step runs on your training server or laptop.
+The key conversion function to call is
+``torch.backends._nnapi.prepare.convert_model_to_nnapi``,
+but some extra steps are required to ensure that
+the model is properly structured.
+Most notably, quantizing the model is required
+in order to run the model on certain hardware backends.
+
+You can copy/paste this entire Python script and run it,
+or make your own modifications.
+By default, it will save the models to ``~/mobilenetv2-nnapi/``.
+
+.. code:: python
+
+    #!/usr/bin/env python
+    import sys
+    import os
+    import torch
+    import torch.utils.bundled_inputs
+    import torch.utils.mobile_optimizer
+    import torch.backends._nnapi.prepare
```

Comment:
Which PyTorch and torchvision versions are required to run this? I built and installed PyTorch from the master branch on Oct 22, got an error "ModuleNotFoundError: No module named 'torch.backends._nnapi'". Then I built from the latest pytorch master branch, and got a new error when running the script:

Exception: Unsupported node kind ('aten::size') in node %13 : int = aten::size(%input.88, %32) # /Users/jeffxtang/opt/anaconda3/lib/python3.8/site-packages/torchvision/models/mobilenet.py:166:0

I then installed torchvision from the latest source but the script still picks the older 0.7...

Reply:
Updated instructions to reference the specific known-working version.